public/app/core/logsModel.ts

```diff
@@ -747,6 +747,63 @@ export function queryLogsVolume<TQuery extends DataQuery, TOptions extends DataS
   });
 }
 
+/**
+ * Creates an observable, which makes requests to get logs samples.
+ */
+export function queryLogsSample<TQuery extends DataQuery, TOptions extends DataSourceJsonData>(
+  datasource: DataSourceApi<TQuery, TOptions>,
+  logsSampleRequest: DataQueryRequest<TQuery>
+): Observable<DataQueryResponse> {
+  logsSampleRequest.hideFromInspector = true;
+
+  return new Observable((observer) => {
```

Comment:
Do we really need to use observables here? I can't find the use case and the produced code is very obscure.